### PR TITLE
Allow custom public ids for uploaded cloudinary files

### PR DESCRIPTION
**Pull request by dpikt**
Came up today on empower, where they'd like to override the filename generation for uploaded files.

Note: the check against `DEFAULT_FILE_TYPE` is knowingly removed because it was untested, and any code relying on it (probably none) should be considered a regression.

### CODE REVIEW

File: test/cloudinary-uploader.test.js [hunk 1]
```diff
@@ -83,6 +83,19 @@ test('cloudinaryUploader sets `publicId`', () => {
   })
 })
 
+test('cloudinaryUploader allows custom `publicId` creator', () => {
```
Comment (chawes13) [inline test/cloudinary-uploader.test.js]: Can you add a test that covers the case where `createPublicId` and `cloudinaryPublicId` are both passed?

**Comment by chawes13** [comment]
@dpikt It looks like there is an `lp-hoc` dependency in `lp-components`. For this to work, the client must have `lp-components: ^3.20.3`. Would you mind documenting this in the `v4.2.0` release notes?

**Comment by dpikt** [comment]
@chawes13 this is in the context of using `CloudinaryFileInput`? It might make more sense to document that dependency in `lp-components` instead 🤔

**Comment by chawes13** [comment]
@dpikt Yes, it's related to a hard to diagnose issue that Ifat and I stumbled into. Yeah, documenting it in `lp-components` makes more sense!

**Pull request merged**